File: primes/javascript/primes.js

const times = [];

for (let i = 0; i < 10; i++) {
    console.log(`staring ${i} run`)
    const startTime = Date.now();
    primes(10_000_000)
    const totalTime = Date.now() - startTime;
    times.push(totalTime);
}

const average = times.reduce((a, b) => a + b, 0) / times.length;
console.log(`Average time was ${average}ms`)



function primes(max) {
    const primesList = [];

    primesList.push(2);

    for (let i = 3; i < max; i++) { //go through all numbers
        let isPrime = true;
        for (let j = 0; j < primesList.length; j++) { //test each number
            let prime = primesList[j];
            if (prime > Math.sqrt(i)) {
                break;
            }
            if (i % prime === 0) { //i = test number, j = some known prime
                isPrime = false;
                break;
            }
        }

        if (isPrime) {
            primesList.push(i);
        }
    }

    return primesList;
}

File: primes/python/primes.py
import time
from math import sqrt


def primes(max):
    primes_list = [2]

    for i in range(3, max):
        is_prime = True
        for prime in primes_list:
            if prime > sqrt(i):
                break
            if i % prime == 0:
                is_prime = False
                break

        if is_prime:
            primes_list.append(i)

    return primes_list


start_millis = round(time.time() * 1000)
primes(10_000_000)
total_millis = round(time.time() * 1000) - start_millis

print(total_millis)
